src/components/data/RenameTableModal.js

```diff
@@ -1,26 +1,50 @@
 import _ from 'lodash/fp'
 import { Fragment, useState } from 'react'
 import { div, h } from 'react-hyperscript-helpers'
-import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common'
-import { allSavedColumnSettingsEntityTypeKey, useSavedColumnSettings } from 'src/components/data/SavedColumnSettings'
+import { ButtonPrimary, IdContainer, RadioButton, spinnerOverlay } from 'src/components/common'
+import { warningBoxStyle } from 'src/components/data/data-utils'
+import { allSavedColumnSettingsEntityTypeKey } from 'src/components/data/SavedColumnSettings'
+import { icon } from 'src/components/icons'
 import { ValidatedInput } from 'src/components/input'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
 import { withErrorReporting } from 'src/libs/error'
 import Events from 'src/libs/events'
 import { FormLabel } from 'src/libs/forms'
 import * as Utils from 'src/libs/utils'
 
 
-const RenameTableModal = ({ onDismiss, onUpdateSuccess, namespace, name, selectedDataType, entityMetadata }) => {
+const RenameTableModal = ({ onDismiss, onUpdateSuccess, getAllSavedColumnSettings, updateAllSavedColumnSettings, setTableNames, namespace, name, selectedDataType }) => {
   // State
   const [newName, setNewName] = useState('')
   const [renaming, setRenaming] = useState(false)
+  const [renameSetTables, setRenameSetTables] = useState(false)
 
-  const {
-    getAllSavedColumnSettings,
-    updateAllSavedColumnSettings
-  } = useSavedColumnSettings({ workspaceId: { namespace, name }, entityType: selectedDataType, entityMetadata })
+  const handleTableRename = async ({ oldName, newName }) => {
+    Ajax().Metrics.captureEvent(Events.workspaceDataRenameTable, { oldName, newName })
+    await Ajax().Workspaces.workspace(namespace, name).renameEntityType(oldName, newName)
+  }
+
+  const moveTableColumnSettings = async tableNames => {
```

Comment:
I don't understand what this code is doing (not knowing much about data). It would be awesome to have a reviewer form AJ, though I'm sure Nick already took a close look.

Reply:
Ah, and I see it mostly just moved up from below.